broker/src/main/java/io/zeebe/broker/clustering/atomix/ZeebeRaftStateMachine.java

```diff
@@ -64,7 +64,7 @@ public ThreadContext executor() {
       if (index > reader.getFirstIndex()) {
         final var future = new CompletableFuture<Void>();
         logger.debug("Compacting log up from {} up to {}", reader.getFirstIndex(), index);
-        raft.getThreadContext().execute(() -> safeCompact(index, future));
+        threadContext.execute(() -> safeCompact(index, future));
```

Comment:
Do we still need this one as well?

Reply:
You are right. It is also always called from raft context.